cranelift-codegen/meta/src/isa/x86/recipes.rs

```diff
@@ -3035,10 +3036,87 @@ pub(crate) fn define<'shared>(
     recipes.add_recipe(
         EncodingRecipeBuilder::new("safepoint", &formats.multiary, 0).emit(
             r#"
-            sink.add_stackmap(args, func, isa);
-        "#,
+                sink.add_stackmap(args, func, isa);
+            "#,
         ),
     );
 
+    /*recipes.add_recipe(
+        EncodingRecipeBuilder::new("tls_value", &formats.unary_global_value, 19)
+            .operands_out(vec![gpr])
+            .emit(
+                r#"
+                    // output %rax
+                    // clobbers %rdi
+
+                    // leaq @tlsld(%rip), %rdi
+                    sink.put1(0x48);
+                    sink.put1(0x8d);
+                    sink.put1(0x3d);
+                    sink.reloc_external(Reloc::ElfX86_64TlsLd,
+                                        &func.global_values[global_value].symbol_name(),
+                                        -4);
+                    sink.put4(0);
+
+                    // callq __tls_get_addr-4
+                    sink.put1(0xe8);
+                    sink.reloc_external(Reloc::X86CallPLTRel4,
+                                        &ExternalName::LibCall(LibCall::ElfTlsGetAddr),
+                                        -4);
+                    sink.put4(0);
+
+                    // leaq @dtpoff32(%rax), %rax
+                    sink.put1(0x48);
+                    sink.put1(0x8d);
+                    sink.put1(0x80);
+                    sink.reloc_external(Reloc::ElfX86_64DtpOff32,
+                                        &func.global_values[global_value].symbol_name(),
+                                        0);
+                    sink.put4(0);
+                "#,
+            ),
+    );*/
+
+    recipes.add_recipe(
+        EncodingRecipeBuilder::new("elf_tlsld", &formats.unary_global_value, 7)
+            // FIXME Should use gpr instead of reg_rdi, but that causes an unnecessary regmove instruction.
+            .operands_out(vec![reg_rdi])
+            .emit(
+                r#"
+                    // leaq @tlsld(%rip), %rdi
+                    sink.put1(0b0100_1000);
+                    const LEA: u8 = 0x8d;
+                    sink.put1(LEA);
+                    modrm_riprel(0b111 /*out_reg0*/, sink);
```

Comment:
From looking up `0b111` it seems that this is using `%rdi` (from your comments I take it that that is a requirement to make this work?) but we might want to make that more clear here.

Reply:
Technically it isnt, but it causes suboptimal codegen otherwise. Also I will change this to move to the GD tls model, which requires a specific sequence of prefixes and insts in one block.

Reply:
It doesn't seem future-proof to add register allocation constraints because regalloc isn't doing the best job right now. We're working on replacing the current regalloc by something simpler in the future. Please either make it a TODO and open an issue about it (when this lands), or remove the constraint.